src/main/java/com/coveo/spillway/storage/sliding/InMemorySlidingStorage.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * The MIT License
+ * Copyright (c) 2016 Coveo
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.coveo.spillway.storage.sliding;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import com.coveo.spillway.limit.LimitKey;
+import com.coveo.spillway.storage.InMemoryStorage;
+import com.coveo.spillway.storage.LimitUsageStorage;
+import com.coveo.spillway.storage.utils.AddAndGetRequest;
+import com.coveo.spillway.storage.utils.Capacity;
+
+/**
+ * Implementation of {@link LimitUsageStorage} using memory.
+ * <p>
+ * Not recommended as a distributed storage solution since sharing memory
+ * can be complicated. Perfect for local usages.
+ * <p>
+ * The difference with {@link InMemoryStorage} is that instead of fixed time
+ * buckets, this implementation uses a sliding window of time and counts
+ * the number of requests made in that window.
+ * This implementation CANNOT be used as wrapped storage for {@link AsyncSlidingLimitUsageStorage}.
+ *
+ * @author Emile Fugulin
+ * @since 1.1.0
+ */
+public class InMemorySlidingStorage implements LimitUsageStorage {
+  private Map<Instant, Map<LimitKey, Capacity>> map = new ConcurrentHashMap<>();
+  private Clock clock = Clock.systemDefaultZone();
+  private Duration retention;
+  private Duration slideSize;
```

Comment:
this variable isn't the entire slidingWindow size, its the size of a division in that window.